### PR TITLE
implement sdk-merkle prepare for Events System

### PR DESCRIPTION
**Pull request by shekohex**
This PR is for pre-adding `Events Decoding` in the `app-utils` package.

Any MerkleTree functionality (create, add a leaf, add leaves, root, generate zkproof, ..etc) all are moved from the `sdk-mixer` and are now lives in the `sdk-merkle` this would help to add the events system.

that means, in the Mixer, when a user deposit, we only need `sdk-mixer`, they don't need any interaction with `sdk-merkle` and the same goes when we do a withdrawal, we would use `sdk-merkle`

### CODE REVIEW

File: packages/sdk-mixer/src/mixer.ts [hunk 1]
```diff
@@ -55,7 +52,7 @@ export class Mixer extends EventBus<MixerEventMap> {
       };
       // this will be triggered if the worker rasid the event restart so we can  reject the promises
       off = this.on('restart', () => {
-        reject();
+        reject(new Error('Restarted'));
         // if this is called that means the worker has events (work)that aren't resolved
```
Comment (drewstone) [inline packages/sdk-mixer/src/mixer.ts:56]:
```suggestion
        // if this is called that means the worker has events (work) that aren't resolved
```

**Review by drewstone — APPROVED** [review]
This looks good! I would even suggest including examples in `scripts` for using the mixer especially. Calling `deposit` `withdraw` with demonstrations of how this calls into the merkle tree and more. What do you think @shekohex ?

**Comment by shekohex** [comment]
Yes, in the next PR for adding Events decoding, there would be examples/scripts for quick deposit/withdrawal to demo the new events too.

**Comment by drewstone** [comment]
Are there corresponding PRs that need to be put up in the dApp to complement this?

**Pull request merged**